cmake/pcl_options.cmake

```diff
@@ -66,7 +66,12 @@ option(BUILD_tools "Useful PCL-based command line tools" ON)
 option(WITH_DOCS "Build doxygen documentation" OFF)
 
 # set index size
-set(PCL_SIZE_INDEX 32 CACHE STRING "Set index size. Available options are: 8 16 32 64")
+set(PCL_INDEX_SIZE 32 CACHE STRING "Set index size. Only available from PCL 1.12 or greater. Available options are: 8 16 32 64")
```

Comment:
If the user didn't specify a default, use the behavior, else pick up the custom value. My code was for the default value, not for the final value

Reply:
I'm not sure I follow. Isnt that what's happening now? By default, the value is 32. But the user may change it if he wishes. As long as the version is greater than 1.11, the value is going to be 32 by default (or whichever value the user picks, if he changes it)

I didn't get what you meant by this:
> use the behavior

Reply:
We were using `int` for indices before I shifted to constant size integer type. However, that has a problem: namely windows uses 64 bit integer, while others use 32 bit. As a result, to maintain compatibility, till PCL 1.12, indices would be `int` after which they'd be int32 uniformly. This is the default behavior.

The user should be allowed to choose a custom size no matter PCL version. In your current code, if the uses chooses 64, they'd still get 32 on version < 1.12

What I meant to say was:
* Keep the default behavior unless user specifies a custom type (currently `int` for PCL 1.11 and `int32` for PCL 1.12)
* Respect the user's custom wishes (even on PCL 1.11)

Reply:
Thanks for the clarification. To support this functionality, we would need to use an additional cmake option to allow the user to decide whether to use custom indices or not, in addition to the size selection right? For instance something like
```cmake
set(PCL_INDEX_CUSTOM false CACHE BOOL "Set whether custom types should be allowed for indices. Disabled by default.")
if(PCL_INDEX_CUSTOM)
  set(PCL_INDEX_SIZE 32 CACHE STRING "Set index size. Available options are: 8 16 32 64")
  set(PCL_INDEX_CUSTOM_STR "true")
else()
  set(PCL_INDEX_CUSTOM_STR "false")
endif()
```
This would basically require two levels of configuration with ccmake right? First to enable custom types and then to select the custom size. Would that be fine?

Reply:
You don't need 2 steps for the use. This can be done in 2 ways behind the scenes:

1. Have the default value be empty or something bad (like -1). In C++ macro, check if PCL_INDEX_SIZE is invalid, and if it is, then use the default behavior, else user's choice
2. Have a macro in C++, PCL_INDEX_SIZE_DEFAULT. Let the user set the value of PCL_INDEX_SIZE which is a non-integer PCL_INDEX_SIZE_DEFAULT by default. I haven't tested in in CMake, but this works for Makefile and commandline.